projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/JobUploadValidator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.upload;
+
+import com.vmware.taurus.exception.ExternalSystemError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+@Service
+class JobUploadValidator {
+  private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+  /**
+   * List of files that are allowed to be uploaded. Full list of file types are documented in
+   * https://tika.apache.org/2.5.0/formats.html#Full_list_of_Supported_Formats_in_standard_artifacts
+   * If set to empty, then all files are allowed using the allowlist.
+   */
+  private final String[] allowlist;
+
+  private final FileFormatDetector formatDetector;
+
+  public JobUploadValidator(
+      @Value("${upload.validation.fileTypes.allowlist:}") String[] allowList) {
+    this.allowlist = allowList;
+    this.formatDetector = new FileFormatDetector();
+    log.debug("Job sources upload validator allowlist: {}", Arrays.toString(allowList));
+  }
+
+  /**
+   * Validate all files of a data job that are being upto standard and are not containing anything
+   * forbidden. The validation done is by detecting the file type and checking if that file type is
+   * allowed against preconfigurd list controlled by "upload.validation.fileTypes.allowlist". If the
+   * allowList is empty then all files are allowed.
+   *
+   * @param jobName the name of the data job whose files are validated
+   * @param jobDirectory path to the data job directory where unarchived content of the data job
+   *     being uploaded can be seen.
+   * @throws InvalidJobUpload
+   */
+  public void validateJob(String jobName, Path jobDirectory) throws InvalidJobUpload {
```

Comment:
Is something else going to happen in this method eventually? 
At the moment all it does is call another method?

Reply:
I suppose not. It remained alone after some refactoring. I like to keep the name because it's clear what the method does. 

I like to keep validateJob as the name of the method since we can have multiple validators and in the future, we can create an interface , and that would be one implementation.